### ISSUE TITLE
Crashes after using barcode scanner

### ISSUE THREAD

**Issue by grisha-v** [aritchie/acr-xamarin-forms#99]
Hello,
I'm trying to use the barcode scanner and it causes Xamarin to crash on Android.
I have some ContentPage (inside NavigationPage) which presents a listview and a button to open the barcode scanner page.
When I open the barcode scanner and press Android back button (without actually scanning any barcode) it returns to the list. After 2 or 3 such attempts, Xamarin crashes with the following exception:
 Unhandled Exception:
 System.ArgumentException: 'jobject' must not be IntPtr.Zero.
 Parameter name: jobject
   at Android.Runtime.JNIEnv.CallVoidMethod (IntPtr jobject, IntPtr jmethod, Android.Runtime.JValue[] parms) [0x00000] in <filename unknown>:0 
   at Android.Widget.AdapterView.set_OnItemClickListener (IOnItemClickListener value) [0x00000] in <filename unknown>:0 
   at Xamarin.Forms.Platform.Android.ListViewAdapter.Dispose (Boolean disposing) [0x00000] in <filename unknown>:0 
   at Java.Lang.Object.Finalize () [0x00000] in <filename unknown>:0 

This is probably connected to this bug:
https://bugzilla.xamarin.com/show_bug.cgi?id=24982
but I can't reproduce it without barcode scanning page, so this must be something in barcode scanning module which causes this crash. Do you have any ideas?
Thanks in advance.

**Comment by aritchie** [comment]
This exception and the defect you have listed have absolutely nothing to do with this library or zxing.  It is likely you have a memory leak in code elsewhere.

**Comment by grisha-v** [comment]
Hello,
I succeeded to reproduce this issue in a sample project (can be downloaded from http://www.filedropper.com/listtest )

and sent it to Xamarin support. They definitely say that the memory leak is inside barcode scanning page.

Please find attached the answer I got from them:

Prashant Cholachagudd
MAR 25, 2015  |  07:36AM EDT
Hi Grisha,

I have profiled your sample app, the barcode scanner is leaking the memory every time I try to read a barcode the allocation starts to pile up and never freed
(http://screencast.com/t/iXGYdTxX).

You may want to contact XamForms Barcode author to fix the memory leaks

Cheers,
Prashant 

----------------------
Please, rate my reply: http://nicereply.com/xamarin/21048876/135521
Grisha 
MAR 24, 2015  |  04:17AM EDT 
Hi Prashant,

After removing most parts of my actual project, I succeeded to build some test project that reproduces the issue.

Please find attached.

It consists of two almost identical pages with list and “scan barcode” button, so that tapping a list record switches between pages.

To reproduce the issue, one need to tap “scan barcode” button, then hardware back button to return to list, then tap list cell to switch to the second page, “scan barcode” there and back to list. After 3-4 such cycles, the application crashes…

Hope this helps.

Best regards,

Grisha.

**Comment by cloush** [comment]
Hi grisha-v,

we had the same problems with the ZXing barcode scanner.
We solved it be creating the Controls for the page in the OnAppearing Method and setting the Content to null in OnDisappearing.

A detailed description can be found on [blog.cloush.com](http://blog.cloush.com/?p=12)

**Comment by grisha-v** [comment]
Hi cloush,
Thanks for the suggestion. I tried your approach, but unfortunately in my case it didn't help.
Even when creating all the controls in OnAppearing and setting them to null in OnDisappearing, the app crashes...

On the contrary to what you've written in your blog, I succeeded to reproduce this bug in a small project pretty easily. The link can be found above.

(Ouch... The file was deleted somehow. I'm trying to use another upload service now:
http://s000.tinyupload.com/index.php?file_id=20120254684748633410)

Anyway, Xamarin support guy suggested that this issue results from memory leak inside the barcode scanner. So, since this issue was closed, I opened another one, #104. I hope Allan Ritchie will be able to help us with this.
Thanks anyway.
